crates/rattler_shell/src/activation.rs

```diff
@@ -433,7 +433,16 @@ impl<T: Shell + Clone> Activator<T> {
         let activation_script_path = activation_script_dir
             .path()
             .join(format!("activation.{}", self.shell_type.extension()));
-        fs::write(&activation_script_path, &activation_detection_script)?;
+
+        // Use CRLF line endings on Windows batch scripts
+        if self.shell_type.extension() == "bat" {
+            activation_detection_script = activation_detection_script.replace('\n', "\r\n");
```

Comment:
For some reason this is totally necessary.

Reply:
Ideally we make this part of the Shell type instead of checking the extension of the Shell type.

Reply:
Like Shell::line_endings or something.